netbox/utilities/conversion.py

```diff
@@ -48,17 +48,17 @@ def to_meters(length, unit):
         raise TypeError(_("Invalid value '{length}' for length (must be a number)").format(length=length))
 
     if unit == CableLengthUnitChoices.UNIT_KILOMETER:
-        return length * 1000
+        return round(Decimal(length * 1000), 4)
     if unit == CableLengthUnitChoices.UNIT_METER:
-        return length
+        return round(Decimal(length), 4)
     if unit == CableLengthUnitChoices.UNIT_CENTIMETER:
-        return length / 100
+        return round(Decimal(length / 100), 4)
     if unit == CableLengthUnitChoices.UNIT_MILE:
-        return length * Decimal(1609.344)
+        return round(length * Decimal(1609.344), 4)
     if unit == CableLengthUnitChoices.UNIT_FOOT:
-        return length * Decimal(0.3048)
+        return round(length * Decimal(0.3048), 4)
     if unit == CableLengthUnitChoices.UNIT_INCH:
-        return length * Decimal(0.0254)
+        return round(length * Decimal(0.0254), 4)
```

Comment:
`to_meters` doesn't explicitly say `length` has to be an `int` (although I suppose it might be enforced earlier by the caller), just a "number", so I expect someone will enter a `float` at some point.

If incoming the `length` is a float, then mile, foot, and inch fail with `TypeError: unsupported operand ...`. The following tests capture it:

```python
        self.assertEqual(
            to_meters(1.0, CableLengthUnitChoices.UNIT_MILE), Decimal('1609.344')
        )
        self.assertEqual(
            to_meters(1.0, CableLengthUnitChoices.UNIT_FOOT), Decimal('0.3048')
        )
        self.assertEqual(
            to_meters(1.0, CableLengthUnitChoices.UNIT_INCH), Decimal('0.0254')
        )
```

Reply:
Perhaps it makes sense to just cast `length` to a `Decimal` as soon as the early validation is done.

Reply:
Good catch!